sigmf/sigmffile.py

```diff
@@ -201,18 +201,23 @@ def __next__(self):
             raise StopIteration
 
     def __getitem__(self, sli):
-        a = self._memmap[sli] # matches behavior of numpy.ndarray.__getitem__()
-        if self._return_type is not None:
-            # is_fixed_point and is_complex
-            if self._memmap.ndim == 2:
-                # num_channels == 1
-                a = a[:,0].astype(self._return_type) + 1.j * a[:,1].astype(self._return_type)
-            elif self._memmap.ndim == 3:
-                # num_channels > 1
-                a = a[:,:,0].astype(self._return_type) + 1.j * a[:,:,1].astype(self._return_type)
-            else:
-                raise ValueError("unhandled ndim in SigMFFile.__getitem__(); this shouldn't happen")
-        return a
+        mem = self._memmap[sli] # matches behavior of numpy.ndarray.__getitem__()
+        
+        if self._return_type is None:
+            return mem
+
+        a = np.reshape(mem, (-1, 2)) #handles case where sli is an int so a is 1D  
```

Comment:
```suggestion
        ray = np.reshape(mem, (-1, 2)) #handles case where sli is an int where ray is 1D  
```
single letter variables bad